packages/core/src/plugins/Plugin.ts

```diff
@@ -0,0 +1,25 @@
+import type { DependencyManager } from './DependencyManager'
+
+import { injectable } from 'tsyringe'
+
+export interface Plugin {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  new (...args: any[]): any
+  register(dependencyManager: DependencyManager): void
+}
+
+/**
+ * Decorator that marks the class as a plugin. Will enforce the required interface for a plugin (with static methods)
+ * on the class declaration. A plugin doesn't have a public api by default, if you need this, use the `@modulePlugin` decorator.
+ */
+export function plugin() {
```

Comment:
Nice! However, very optional, which do you prefer? It just popped into my head and might be worth considering.

```ts
@pluginModule()
export class modulePlug() {}

// or

enum PluginToken = {
  None = 0,
  Module = 1
}

// ...

@plugin(PluginToken.Module)
export class modulePlug() {}
```

I slightly prefer the second one but I am fully down to use the first one aswell.

Reply:
I think my preference goes to the first one. The API feels simpler. Typing would also be more complex if the interface for a module and normal plugin drift apart (now we can just type the method, then we will need to work with complex generics based on the token you pass to determine what the interface should be)

Reply:
@karimStekelenburg @morrieinmaas what's your preference?

Reply:
I'm in favour of the first approach. Apart from making the typing easier (I'd definitely like to avoid adding more complex generics magic) I think it would be clearer from a documentation perspective.

Reply:
Solved as we now only have `@module()`